xla/service/gpu/buffer_comparator.cc

```diff
@@ -182,49 +183,56 @@ static absl::StatusOr<bool> CompareEqualParameterized(
 absl::StatusOr<bool> BufferComparator::CompareEqual(
     se::Stream* stream, se::DeviceMemoryBase current,
     se::DeviceMemoryBase expected) const {
+
+  ComparisonParams params{
+    .relative_tol = relative_tol_,
+    .verbose = verbose_,
+    .shape = shape_, // Keep only reference to the shape
+    .stream = stream,
+    .current = current,
+    .expected = expected,
+  };
```

Comment:
Named initializers are a C++20 construct and XLA is c++17, so this will fail our internal presubmit, can you change it please?